imap_data_access/file_validation.py

```diff
@@ -1,3 +1,4 @@
+# ruff: noqa: PLR0913
 """Methods for managing and validating filenames and filepaths"""
```

Comment:
I found an issue while updating lambda to use this new validation method. Can you update `extract_filename_components`
from this
```
pattern = (
            r"^(?P<mission>imap)_"
            r"(?P<instrument>[^_]+)_"
            r"(?P<datalevel>[^_]+)_"
            r"(?P<descriptor>[^_]+)_"
            r"(?P<startdate>\d{8})_"
            r"(?P<enddate>\d{8})_"
            r"(?P<version>v\d{2}-\d{2})"
            r"\.(?P<extension>cdf|pkts)$"
        )
```
to this
```
pattern = (
            r"^(?P<mission>imap)_"
            r"(?P<instrument>[^_]+)_"
            r"(?P<data_level>[^_]+)_"
            r"(?P<descriptor>[^_]+)_"
            r"(?P<start_date>\d{8})_"
            r"(?P<end_date>\d{8})_"
            r"(?P<version>v\d{2}-\d{2})"
            r"\.(?P<extension>cdf|pkts)$"
        )
```
This way it matches database naming.